### ISSUE TITLE
Cannot package native extension with macOS implementation without native code

### ISSUE THREAD

**Issue by itlancer** [Gamua/Adobe-Runtime-Support#754]
## Problem Description
Cannot package native extension with macOS implementation without native code.
Sometimes when developing native extensions necessary to make implementations for some platforms **without** native code. And it works fine for Windows, Android, iOS and tvOS but cause error when trying to package such implementation for macOS.

It has been tested with multiple AIR SDK versions, even with latest AIR 33.1.1.406 with Windows and macOS devices.
The same problem in all cases.

## Steps to Reproduce
Try to package any native extension with macOS implementation **without** native code. 
`adt -package -storetype pkcs12 -keystore cert.p12 -storepass pass -keypass pass -target ane TestANE.ane extension.xml -swc default/TestANE.swc -platform MacOS-x86-64 -C macos . -platform default -C default .`

Native extension example with source code and packaging resources attached.
[macos_ane_without_native_code_bug.zip](https://github.com/Gamua/Adobe-Runtime-Support/files/6216002/macos_ane_without_native_code_bug.zip)

**Actual Result:**
Native extension packaging via `adt` cause error:
```
unexpected failure: null
java.lang.NullPointerException
        at com.adobe.air.ANEPackager.validate(ANEPackager.java)
        at com.adobe.air.ANEPackager.createPackage(ANEPackager.java)
        at com.adobe.air.ADT.parseArgsAndGo(ADT.java)
        at com.adobe.air.ADT.run(ADT.java)
        at com.adobe.air.ADT.main(ADT.java)
```

**Expected Result:**
Native extension will be correctly packaged and can be used for macOS with provided implementation without native code (only AS3 code).

## Known Workarounds
1) Provide any native code for macOS implementation even if you don't need it.
2) Make macOS "implementation" in application itself, so such native extension for macOS useless and cause "dirty" code if you target for multiple platforms.
3) Make macOS "implementation" in default implementation.

**Comment by marchbold** [comment]
Just out of interest why would you have an ANE platform with no native code? I would have thought you'd use the default implementation for that.

**Comment by itlancer** [comment]
@marchbold, thanks for idea, also added as workaround.
In my cases all applications crossplatform. And sometimes I need to work, for example, with native windows (by native code, when `NativeWindow` functionality not enough) which not available on mobile platforms. So for Android/iOS I made "fallback" implementation with AS3 `DisplayList` (without native code), for Windows use C++ "classic" native extension for that and for macOS made AS3 implementation using `NativeWindow` (cause macOS doesn't support some specific stuff that I need).

In such case all AS3 implementations for Android/iOS/macOS could be differ and each one in it own class (not default) without need to check OS type and other conditions.
This is not blocker but simplify developing and make code clearer in some cases.

**Comment by marchbold** [comment]
Sounds to me like you could switch implementation in the default lib based on the platform? Different class for each platform and a generic interface definition?

**Comment by itlancer** [comment]
@marchbold
Hope I get you right.
Right now my native extension (for example) looks like:
**default**: AS3, used for tvOS (like fallback)
**Windows**: native
**Android**: AS3 (without native)
**iOS**: AS3 (without native)
**macOS**: none (cannot include AS3 without native code cause of this issue)

You suggest (workaround 3):
**default**: AS3, used for tvOS, Android, iOS, macOS with OS detection and code branching depends on platform
**Windows**: native

I would like to have:
**default**: AS3, used for tvOS (like fallback)
**Windows**: native
**Android**: AS3 (without native)
**iOS**: AS3 (without native)
**macOS**: AS3 (without native)

**Comment by marchbold** [comment]
Yes that is correct, I think your usage is not really what a native extension is designed for i.e. it's designed to package "native" implementations. 

I actually feel that attempting to implement a platform without native code should fail!

**Comment by ajwfrost** [comment]
Hi guys. Leaving architectural/strategic views aside, I'm not particularly keen on seeing a NullPointerException causing ADT to abort with an error, so one of the guys put a null-check in.. see the below link. I believe this means that it will still allow a Mac platform to be packaged without needing a native part of the implementation.
https://transfer.harman.com/message/6o095qBtCIdZjFyzEsST98

**Comment by marchbold** [comment]
@ajwfrost Great, definitely an issue, ie not a designed failure :) so glad you are looking into it!

**Comment by itlancer** [comment]
@ajwfrost, thank you very much! It works as I expected. Waiting for release to close this issue.

**Comment by itlancer** [comment]
It fixed with latest AIR SDK 33.1.1.476.
Thank you!
